docs/adr/0002-yarn-v4.md

```diff
@@ -0,0 +1,243 @@
+
+<details>
+  <summary>Tests on hardened mode on "cachi2 fetch-deps" time</summary>
+
+**Test method**: Call `cachi2 fetch-deps` while setting the [enableHardenedMode](https://yarnpkg.com/configuration/yarnrc#enableHardenedMode) `.yarnrc.yml` option.
+
+**plop**<br>
+884 deps - 72M<br>
+https://github.com/plopjs/plop<br>
+e0122279d1376ee62604acbbff1e76a88935b1af
+
+non-hardened:<br>
+37.53s user 9.72s system 321% cpu 14.682 total<br>
+38.00s user 9.37s system 339% cpu 13.957 total<br>
+36.16s user 9.31s system 327% cpu 13.873 total<br>
+*average time 14.17*<br>
+
+hardened:<br>
+45.00s user 13.99s system 290% cpu 20.319 total<br>
+50.67s user 15.45s system 299% cpu 22.099 total<br>
+56.10s user 14.89s system 352% cpu 20.148 total<br>
+*average time 20.855*<br>
+~47% increase
+
+===============================================
+
+**porta**<br>
+1387 deps - 505M
+https://github.com/3scale/porta<br>
+bcbaea6392b58a89291932b2289127e44864e288<br>
+(upgraded to yarn-4.5.3)
+
+non-hardened:<br>
+29.32s user 9.66s system 369% cpu 10.554 total<br>
+32.41s user 10.69s system 406% cpu 10.604 total<br>
+29.43s user 9.97s system 362% cpu 10.864 total<br>
+*average time 10.674*<br>
+
+hardened:<br>
+41.66s user 16.28s system 294% cpu 19.646 total<br>
+65.75s user 19.94s system 416% cpu 20.576 total<br>
+64.89s user 20.47s system 404% cpu 21.106 total<br>
+*average time 20.442*<br>
+*~91% increase*
+
+================================================
+
+**twenty**<br>
+4139 deps - 2.2G<br>
+https://github.com/twentyhq/twenty<br>
+e492efb79e8fd8d8f8292c2fccd991d4448ac249<br>
+
+non-hardened:<br>
+122.67s user 75.31s system 305% cpu 64.91 total<br>
+173.75s user 92.09s system 336% cpu 78.99 total<br>
+111.99s user 69.36s system 315% cpu 57.568 total<br>
+*average time 67.155*
+
+hardened:<br>
+245.61s user 103.60s system 351% cpu 99.30 total<br>
+274.81s user 109.91s system 361% cpu 106.42 total<br>
+244.61s user 107.89s system 349% cpu 100.81 total<br>
+*average time 102.176*<br>
+*~52 increase*
+
+</details>
+
+## Decision
+
+The changes needed to support Yarn v4 boil down to raising the maximum supported version, adding a few integration tests to cover v4 scenarios and updating the documentation. The Corepack shim is compatible with v4, so the project will still be processed with the exact Yarn version that is defined in its configuration files.
+
+Enabling the newly introduced hardened mode during the prefetch might prove useful to further increase the security of the process, but is by no means necessary to introduce basic support for v4. The impact on the prefetching speed is something that needs to be investigated in depth before we decide to enable it, but the decision can be deferred to a later moment.
```

Comment:
Could we just add a user-accessible toggle switch for that?

Reply:
We most definitely could, I just wonder if it is justifiable to have the added complexity, and if we would want a user to disable it at all.

If we went that route, we would probably need to write in the SBOM that hardened mode was off, so that a pipeline could gate builds made with hardened mode off.

Reply:
Fair enough, thank you.